### PR TITLE
[dtest] Run tests on an existing cluster launched with docker compose

### PR DESCRIPTION
**Pull request by vpranckaitis**
**What this PR does / why we need it**:

Updates dtest to run tests on already existing docker cluster rather than building and launching it from the Go code. This allows using docker compose for container management, which is more convenient than doing the same in the dtest code. 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note
NONE
```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note
NONE
```

### CODE REVIEW

File: src/cmd/tools/dtest/docker/harness/resources/harness.go [hunk 1]
```diff
@@ -77,20 +77,24 @@ func SetupSingleM3DBNode(opts ...SetupOptions) (DockerResources, error) { // nol
...
+	if !options.existingCluster {
+		err = setupNetwork(pool)
+		if err != nil {
+			return nil, err
+		}
+
+		err = setupVolume(pool)
+		if err != nil {
+			return nil, err
+		}
```
Comment (linasm) [inline src/cmd/tools/dtest/docker/harness/resources/harness.go:90]: Nit:
```suggestion
		if err := setupNetwork(pool); err != nil {
			return nil, err
		}

		if err := setupVolume(pool); err != nil {
			return nil, err
		}
```

**Comment by linasm** [comment]
@gediminasgu can you take this one, I believe you have some recent experience in this area?

File: src/cmd/tools/dtest/docker/harness/resources/common.go [hunk 2]
```diff
@@ -117,6 +111,7 @@ func setupNetwork(pool *dockertest.Pool) error {
 
 	for _, n := range networks {
 		if n.Name == networkName {
+			return nil
```
Comment (gediminasgu) [inline src/cmd/tools/dtest/docker/harness/resources/common.go:114]: left over?

**Review by gediminasgu — APPROVED** [review]
One comment but in general looks good.

File: src/cmd/tools/dtest/docker/harness/resources/common.go (hunk 2 above)
Comment (vpranckaitis) [reply src/cmd/tools/dtest/docker/harness/resources/common.go]: https://github.com/m3db/m3/pull/3067/commits/8543d3a4b1a11116b2cdec2e36be0e6167a0909a

**Pull request merged**